design/004-IoT-Adoption-Watcher.md

```diff
@@ -308,20 +319,90 @@ discard the older ones.
    Relevant parent ledger channel is the ledger channel between this user and
    the common intermediary.
 
+### Illustrative usage of watcher API
+
+The main component can either initialize a local watcher or connect to a remote
+watcher.
+
+#### Using the watcher from the main component
+
+```go
+// Either initialize a local watcher
+w, err := NewWatcher(rs RegistererSubscriber)
+
+// Or connect to the remote watcher
+w, err := ConnectToWatcherGrpc(cfg Config)
+
+// After establishing a channel (say "ch1"),
+// 1. Setup the subscription,
+// 2. Instruct the watcher to start watching for events on that channel and,
+// 3. Publish the initial state.
+statesPub, statesSub := newStatesPubSub()
+adjudicatorEventsPub, adjudicatorEventsSub := newAdjudicatorEventsPubSub()
+
+err := w.StartWatching(ch1.ID(), statesSub, adjudicatorEventsPub)
+
+statesPub.Publish(ch.StateWithSignatures())
```

Comment:
You write that the reason for initializing the pub subs external of the watcher is that you want to be able to pass them to other watchers later.

Another way of supporting concurrent watchers, would be to implement a concurrent watcher type, that internally manages a set of watchers `[w1, w2, ...]`. That is, when you call `w.StartWatching`, it calls `w1.StartWatching`, `w2.StartWatching`, ... . I think it might be more intuitive to use. You could even keep all functionality within the watcher component.

```suggestion
err := w.StartWatching(ch1.ID())
w.PublishState(ch1.ID(), ch1.StateWithSignatures())
e, err := w.NextEvent(ch1.ID())
```

Reply:
1. I like your suggestion on supporting concurrent watching by implementing a concurrent watcher.
So, I have moved the responsibility of setting up pub-sub interfaces to the watcher. The pub-sub interfaces are now returned by the `StartWatching` API. This is updated in b6a2ddd

2. However, I think we should use `statesPub.Publish(ch.StateWithSignatures())` instead of `w.PublishState(ch1.ID(), ch1.StateWithSignatures())`Because publishing a state is done in the context of the channel and for a given channel, the API should allow publishing states only for that channel.

You could see the last section of updated proposal to see how the API works now.